### ISSUE TITLE
Openstack volume attachment not working

### ISSUE THREAD

**Issue by michalmedvecky** [hashicorp/terraform#12000]
### Terraform Version

Terraform v0.9.0-dev (89acb5811a25e59e6a9d55cf5a2429846e3b5982+CHANGES)

### Affected Resource(s)

- openstack_blockstorage_volume_attach_v2

### Terraform Configuration Files
```hcl
resource "openstack_blockstorage_volume_v2" "tftestvol" {
  name = "tf-test-volume"
  description = "Delete me whenever you need"
  size = 100
}
resource "openstack_blockstorage_volume_attach_v2" "tftestmount" {
  instance_id = "${openstack_compute_instance_v2.testinstance.id}"
  volume_id = "${openstack_blockstorage_volume_v2.tftestvol.id}"
  device = "/dev/vdd"
  attach_mode = "rw"
}
```

### Expected Behavior

Volume should be attached to the instance

### Actual Behavior
Volume is somehow semi-attached and cannot be seen by the instance

### Steps to Reproduce
1. `terraform apply`

### Important Factoids

I have two volumes in the same tenants, one of them created/attached by TF and the other one via Horizon.

This is what I see:

```
$ openstack volume list
+--------------------------------------+--------------------+--------+------+-----------------------------------------------------------+
| ID                                   | Display Name       | Status | Size | Attached to                                               |
+--------------------------------------+--------------------+--------+------+-----------------------------------------------------------+
| 913e680e-0066-4f28-ba12-edf1ec57ea68 | tf-test-volume     | in-use |  100 | Attached to mytestinstance.com on /dev/vdd  |
| 6b8ac3dd-a141-449c-adda-3d1d91525b47 | test               | in-use |  100 | Attached to mytestinstance.com on /dev/vdb  |
+--------------------------------------+--------------------+--------+------+-----------------------------------------------------------+
```

`tf-test-volume` is TF created/managed
`test` is Horizon created/attached

But when I run

```
$ nova volume-attachments mytestinstance
+--------------------------------------+----------+--------------------------------------+--------------------------------------+
| ID                                   | DEVICE   | SERVER ID                            | VOLUME ID                            |
+--------------------------------------+----------+--------------------------------------+--------------------------------------+
| 6b8ac3dd-a141-449c-adda-3d1d91525b47 | /dev/vdb | aaba3514-d079-45db-a79c-dfdd13825046 | 6b8ac3dd-a141-449c-adda-3d1d91525b47 |
+--------------------------------------+----------+--------------------------------------+--------------------------------------+
```

then the only horizon-attached volume is shown.

In reality, instance does see only one extra volume on /dev/vdb, and no other.

**Comment by jtopjian** [comment]
@michalmedvecky confirmed. 

It seems that the implementation of the BlockStorage attach call is incomplete. I'm not sure how that happened, but I will look into it.

In the meantime, `openstack_compute_volume_attach_v2` will definitely work.

**Comment by jtopjian** [comment]
tl;dr: definitely use `openstack_compute_volume_attach_v2` to attach a Block Storage volume to a Compute virtual machine.

This is entirely my fault. I had misunderstood the BlockStorage attach call. Currently the `openstack_blockstorage_volume_attach_v2` resource doesn't actually perform any attachments at all. It's simply one step in a volume provisioning phase. The fact that no volume is ever attached to the virtual machine was entirely missed through testing, which adds a second egg on my face.

However, there is some good news. I've been able to redesign `openstack_blockstorage_volume_attach_v2` to provide general purpose volume attachments, even outside of OpenStack. For example, the ability to provision an iSCSI target to be attached to a bare metal server or a virtual machine in some other cloud.

Doc updates and PR on the way.

**Comment by stack72** [comment]
Closed via #12071 

Will be part of Terraform 0.8.8
